Kajabi/sage-lib
# feat(typography): Body font selector scoping

## Description
This PR updates the body selector in `packages/sage-assets/lib/stylesheets/core/_typography.scss` to only apply if the `sage-excluded` class is not set on `body`. Then when styles are desired in a scoped context the `.sage-scoped` class can be used on the closer container.

This will prevent the `Inter` font from applying to the body which is helpful when using Sage on the member side or other environments. It also ensures any global tag selectors are moved and reconciled in the global reboot file so that core files can be included in a project independently as needed.

We are not documenting this on the docs site as this is not a fully robust solution that we want to publicize yet. Build team is using it in a limited context for the Quizzes redesign effort underway.

## Screenshots
|  Class Applied  |  Class Not Applied  |
|--------|--------|
|<img width="1792" alt="Class Applied" src="https://user-images.githubusercontent.com/7331038/151413034-55457cee-a29d-4944-84b6-ed6c31960481.png">|<img width="1792" alt="Class Not Applied" src="https://user-images.githubusercontent.com/7331038/151413046-378a4660-601c-41cc-9e05-26eac0bfef51.png">|

## Testing in `sage-lib`
1. Open http://0.0.0.0:4000/
2. Ensure that the body font is still `Inter` or `Inter var`.

## Testing in `kajabi-products`
1. In Admin, open the developer tools
2. Inspect the body element
3. Add the `sage-excluded` class to the body element and ensure that `Inter var` and/or `Inter` does not get applied. Add `sage-scoped` at a lower level container to see Sage type apply therein.

## Notes
(LOW) Adds some MVP scoping options for Sage to be used in scoped member-facing contexts.
_NOTE: Sage Engineering has reviewed this work in a number of `kp` contexts with the Bridge on and found no regressions._

## Related
https://kajabi.atlassian.net/browse/SAGE-156
